### ISSUE TITLE
async support + async test-project

### ISSUE THREAD

**Issue by benjaoming** [django-wiki/django-nyt#130]
This is a pretty hefty issue to report.

Firstly, async support should of course be part of a notifications application :100: 

There's probably a lot of work to do to get there. I would suggest starting with a copy of test-project, let's call it `test-project-async` and start from there.

**Comment by oscarmcm** [comment]
How are we going to handle if the user doesn't want to use async stuff? are we going to follow the same Django approach (prefix all the async methods with an `a` and then from the now async method call the async one but using `async_to_sync`?

**Comment by oscarmcm** [comment]
How this affects `Django-Wiki` in the end?

**Comment by benjaoming** [comment]
@oscarmcm yes, good question, I can't remember the conclusion but I think I saw a discussion about this in django-ratelimit, specifically which prefix to use for the new functions.

> How this affects Django-Wiki in the end?

It should not affect django-wiki or any other deployments. I think that's a requirement that we need to follow. I can see django-nyt being very useful for other projects down the road... but it doesn't look good if current users are disturbed or async becomes a requirement. I like the simplest use-cases to get people started (i.e. no async and only a cron job to send emails).
